### PR TITLE
[network] Add a modular PeerManagerBuilder

### PR DESCRIPTION
**Pull request by agc-quadra**
## Motivation

Make a modular PeerManageBuilder to encapsulate the create/build/start phases associated with instantiating PeerManager

Incremental step towards modularizing network-builder according to #4626
### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

yes

## Test Plan

don't break existing tests

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

**Comment by agc-quadra** [comment]
@davidiw  peermanager builder ready for incremental addition

**Comment by agc-quadra** [comment]
@davidiw Almost there with conflict resolutions

**Comment by agc-quadra** [comment]
@gregnazario

**Comment by agc-quadra** [comment]
@gregnazario  chasing down merge conflicts.

**Comment by agc-quadra** [comment]
@gregnazario One (almost) last merge conflict.

**Comment by agc-quadra** [comment]
@davidiw @gregnazario 

Resolving the last of the self-induced merge conflicts.   This is also the final preliminary step before we can reap the rewards of this stream of changes.

**Comment by agc-quadra** [comment]
@davidiw @gregnazario  Poking for visibility

File: network/src/peer_manager/builder.rs [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+    }
+
+    fn supported_protocols(&self) -> SupportedProtocols {
+        self.direct_send_protocols
+            .iter()
+            .chain(&self.rpc_protocols)
+            .into()
+    }
+
+    fn augment_direct_send_protocols(
```
Comment (gregnazario) [inline network/src/peer_manager/builder.rs:96]: nit: I'd prefer if you used `extend` like the function you're calling inside it

**Pull request closed without merging**